pyuvdata/uvdata/uvdata.py

```diff
@@ -2771,6 +2982,87 @@ def _make_flex_pol(self, raise_error=False, raise_warning=True):
             # With the new array defined and filled, set the attribute equal to it
             setattr(self, name, new_param)
 
+    def convert_to_flex_pol(self):
+        """
+        Convert a regular UVData object into a flex-polarization object.
+
+        This effectively combines the frequency and polarization axis with polarization
+        changing slowest. Saving data to uvh5 files this way can speed up some kinds
+        of data access.
+
+        """
+        if self.flex_spw_polarization_array is not None:
+            raise ValueError("This is already a flex-pol object")
+
+        if not self.flex_spw:
+            self._set_flex_spw()
+            if not self.future_array_shapes:
+                self.channel_width = np.full(self.Nfreqs, self.channel_width)
+            self.flex_spw_id_array = np.zeros(self.Nfreqs, dtype=int)
+
+        new_spw_array = self.spw_array
+        new_flex_pol_array = np.full(self.Nspws, self.polarization_array[0])
+        new_spw_id_array = np.zeros((self.Nfreqs, self.Npols), dtype=int)
+        for pol_ind, pol in enumerate(self.polarization_array):
+            if pol_ind == 0:
+                new_spw_id_array[:, 0] = self.flex_spw_id_array
+            else:
+                for spw in self.spw_array:
+                    new_spw = (
+                        set(range(self.Nspws * self.Npols + 1))
+                        .difference(new_spw_array)
+                        .pop()
+                    )
+                    new_spw_array = np.concatenate((new_spw_array, np.array([new_spw])))
+                    new_flex_pol_array = np.concatenate(
+                        (new_flex_pol_array, np.array([pol]))
+                    )
+                    spw_inds = np.nonzero(self.flex_spw_id_array == spw)[0]
+                    new_spw_id_array[spw_inds, pol_ind] = new_spw
+
+        spw_sort = np.argsort(new_spw_array)
+        self.spw_array = new_spw_array[spw_sort]
+        self.flex_spw_polarization_array = new_flex_pol_array[spw_sort]
+        self.flex_spw_id_array = new_spw_id_array.reshape(
+            self.Nfreqs * self.Npols, order="F"
+        )
+        self.Nspws = self.spw_array.size
+        self.polarization_array = np.array([0])
+        if self.future_array_shapes:
+            freq_array_use = self.freq_array
+        else:
+            freq_array_use = self.freq_array[0, :]
+        self.freq_array = np.tile(freq_array_use, self.Npols)
+        if not self.future_array_shapes:
+            self.freq_array = self.freq_array[np.newaxis, :]
+        self.channel_width = np.tile(self.channel_width, self.Npols)
+        if not self.metadata_only:
+            if self.future_array_shapes:
+                self.data_array = self.data_array.reshape(
```

Comment:
Here I sort of see why we have `order="F"`, because we want to put the blt-axis to be faster-running, but I guess it's still not 100% clear to me.

Reply:
This is just undoing the reshape in the other function. If you use an "F" there you need it here to get back to what we started with.